src/app/pages/test-results-search/components/advanced-search/advanced-search.ts

```diff
@@ -91,13 +93,18 @@ export class AdvancedSearchComponent {
   ) {
   }
 
-  upperCaseAlphaNum(event: any): void {
-    if (typeof event.target.value !== 'string') return;
+  upperCaseAlphaNum(event: InputChangeEventDetail | InputInputEventDetail, field: string): void {
```

Comment:
dont like how your ammending this fuction to do more then for uppercase a value.  can we look at setting  the flag some other way?